src/Microsoft.DotNet.Build.Tasks/Targets/resources.targets

```diff
@@ -23,16 +21,35 @@
           Inputs="$(StringResourcesPath)"
           Outputs="$(IntermediateResOutputFileFullPath)">
     
-    <GenerateResourcesCode 
+    <GenerateResourcesCode
         ResxFilePath="$(StringResourcesPath)" 
-        IntermediateFilePath="$(IntermediateResOutputFileFullPath)" 
-        OutputSourceFilePath="$(ResourceSRPath)" 
+        OutputSourceFilePath="$(IntermediateResOutputFileFullPath)" 
         AssemblyName="$(AssemblyName)" />
 
     <ItemGroup>
+      <!-- The following Compile element has to be included dynamically inside the Target otherwise intellisense will not work -->
+      <Compile Include="$(IntermediateResOutputFileFullPath)">
+          <Visible>true</Visible>
```

Comment:
<Visible> is not going to work. The visual tree is calculated statically at project load, you cannot control the visibility after that.